File: Algorithms/SortingSeaching/SelectionSort.py

def selection_sort(nums):

    for i in range(len(nums)):

        min_index = i

        for j in range(i, len(nums)):

            if nums[j] < nums[min_index]:

                min_index = j

        nums[i], nums[min_index] = nums[min_index], nums[i]

def main():

    nums = [12, 45, 1, 76, 23, 25, 33, 99, 65, 46]

    print(nums)
    selection_sort(nums)
    print(nums)

if __name__ == '__main__':
    main()

File: Algorithms/SortingSeaching/SelectionSort.js
"use strict";

let selectionSort = (nums) => {

    for (let i in nums) {

        let minIndex = i;

        for (let j = i; j < nums.length; j++) {

            if (nums[j] < nums[minIndex]) {

                minIndex = j;

            }

        }

        [nums[i], nums[minIndex]] = [nums[minIndex], nums[i]]

    }

}

let nums = [12, 45, 1, 76, 23, 25, 33, 99, 65, 46];

console.log(nums);

selectionSort(nums);

console.log(nums);
